homeassistant/components/eq3btsmart/config_flow.py

```diff
@@ -0,0 +1,127 @@
+"""Config flow for eQ-3 Bluetooth Smart thermostats."""
+
+import logging
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import BluetoothServiceInfoBleak
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN
+from .schemas import SCHEMA_NAME, SCHEMA_NAME_MAC
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EQ3ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for eQ-3 Bluetooth Smart thermostats."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+
+        self.name = ""
+        self.mac = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
```

Comment:
If user flows we usually enumerate the devices in range and see if they are supported and than offer the user a device to pick.  Switchbot has a good example of this

Reply:
Govee_ble does as well but it for passive devices only so it might not be as good a reference

Reply:
Not sure this is viable in this case, because the thermostats disable their bluetooth radio frequently and most often are not advertising at all when the user creates the entry.

Reply:
Please add a comment to that effect so it doesn't come up in future reviews as thats not a very common case